server/sonar-server/src/main/java/org/sonar/server/ce/ws/TaskFormatter.java

```diff
@@ -144,12 +142,25 @@ public TaskFormatter(DbClient dbClient, System2 system2) {
     return builder;
   }
 
-  private static Ce.Task.Builder setBranch(Ce.Task.Builder builder, String taskUuid, DtoCache componentDtoCache) {
-    componentDtoCache.getBranchName(taskUuid).ifPresent(
+  private static Ce.Task.Builder setBranchOrPullRequest(Ce.Task.Builder builder, String taskUuid, DtoCache componentDtoCache) {
+    componentDtoCache.getBranchKey(taskUuid).ifPresent(
       b -> {
-        builder.setBranch(b);
-        builder.setBranchType(componentDtoCache.getBranchType(taskUuid)
-          .orElseThrow(() -> new IllegalStateException(format("Could not find branch type of task '%s'", taskUuid))));
+        Common.BranchType branchType = componentDtoCache.getBranchType(taskUuid)
+          .orElseThrow(() -> new IllegalStateException(format("Could not find branch type of task '%s'", taskUuid)));
+        switch (branchType) {
+          case LONG:
+          case SHORT:
+            builder.setBranchType(branchType);
+            builder.setBranch(b);
+            break;
+          case PULL_REQUEST:
+            builder.setPullRequest(b);
+            // TODO set pull request title
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fbranch%2Ffirst-class-pull-request&issues=AWGZgq9Iqu6zmfejgj11&open=AWGZgq9Iqu6zmfejgj11)

[ISSUE_KEY]: AWGZgq9Iqu6zmfejgj11